include/zephyr/drivers/spi.h

```diff
@@ -319,10 +304,9 @@ struct spi_config {
 	uint16_t		slave;
 #endif /* CONFIG_SPI_EXTENDED_MODES */
 
-	const struct spi_cs_control *cs;
+	struct spi_cs_control cs;
```

Comment:
This is a break API change and needs to go through the process. If an out of tree driver implements and uses spi_config today, and goes to update zephyr, this will break that code.

I'm 100% fine with this case myself (though @tbursztyka will need to +1 it). But it needs to be announced on the mailing list and at least be brought up in the architecture meeting.

Reply:
If we go this road, I'd also improve isolation of this module. Applications should, in most cases, not play with `cs` field directly. Some helper function would have been useful to prevent a breaking change here.

Reply:
Unfortunately spi_config is a public struct in a public header, so even with helpers there's no guarantees. It's something we need to live with and work with. The process here is really about, making sure the community is ok with it and getting the word out.

That's ok, but does mean a quick blurb in the architecture meeting and an email. Easy